src/wp-includes/class-wp-scripts.php

```diff
@@ -746,17 +746,126 @@ public function add_data( $handle, $key, $value ) {
+	 * @return array Array of script handles.
+	 */
+	private function get_dependents( $handle ) {
+		$dependents = array();
+
+		// Iterate over all registered scripts, finding ones that depend on the script.
+		foreach ( $this->registered as $registered_handle => $args ) {
+			if ( in_array( $handle, $args->deps, true ) ) {
+				$dependents[] = $registered_handle;
+			}
+		}
+		return $dependents;
+	}
+
+	/**
+	 * Get the strategy mentioned during script registration.
+	 *
+	 * @param string $handle The script handle.
+	 * @return string|bool Strategy in script registration, False if not strategy is mentioned..
+	 */
+	private function get_intended_strategy( $handle ) {
+		$script_args = $this->get_data( $handle, 'script_args' );
+		return isset( $script_args['strategy'] ) ? $script_args['strategy'] : false;
+	}
+
+	/**
+	 * Helper function to check if a script has an `after` inline dependency.
+	 *
+	 * @param string $handle Name of the item. Should be unique.
+	 * @return bool True on success, false on failure.
+	 */
+	private function has_after_inline_dependency( $handle ) {
+		return false !== $this->get_data( $handle, 'after' );
+	}
+
+
+	/**
+	 * Check if all of a scripts dependents are deferrable which is required to maintain execution order.
+	 *
+	 * @param string $handle  The script handle.
+	 * @param array $visited An array of already visited script handles used to avoid looping recursion.
+	 * @return bool True if all dependents are deferrable, false otherwise.
+	 */
+	private function all_dependents_are_deferrable( $handle, $visited = array() ) {
+		// If this node was already visited, this script can be deferred and the branch ends.
+		if ( in_array( $handle, $visited, true ) ) {
+			return true;
+		}
+		$visited[]  = $handle;
+		$dependents = $this->get_dependents( $handle );
+		// If there are no dependents remaining to consider, the script can be deferred and the branch ends.
+		if ( empty( $dependents ) ) {
+			return true;
+		}
+
+		// Consider each dependent and check if it is deferrable.
+		foreach ( $dependents as $dependent ) {
+			// If the dependent script is not using the defer strategy, no script in the chain is deferrable.
+			if ( 'defer' !== $this->get_intended_strategy( $dependent ) ) {
+				return false;
+			}
+
+			// Recursively check all dependent.
+			if ( ! $this->all_dependents_are_deferrable( $dependent, $visited ) ) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Get the correct loading strategy for the script.
+	 *
+	 * @param string  $handle Handle name of the script.
+	 * @return string $strategy return the final strategy.
+	 */
+	private function get_eligible_loading_strategy( $handle = '' ) {
+
+		$intended_strategy = $this->get_intended_strategy( $handle );
+		/**
+		 * Handle known blocking strategy scenarios.
+		 *
+		 * Blocking if not a registered handle.
+		 * If the script has an 'after' inline dependency, then it can't be eligible for async or defer.
+		 */
```

Comment:
Tricky one, but [WP inline comment standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-2-multi-line-comments) only include a single asterisk on the first line. 
```suggestion

		/*
		 * Handle known blocking strategy scenarios.
		 *
		 * Blocking if not a registered handle.
		 * If the script has an 'after' inline dependency, then it can't be eligible for async or defer.
		 */
```